src/agents/container_agent.py

```diff
@@ -157,32 +163,50 @@ async def prepare_result_notification(self, request: ACLMessage) -> ACLMessage:
         return response
 
 
-class ContainerAgent(BaseAgent):
-    def __init__(self, jid: str, password: str, slot_manager_agents_jids: Sequence[SlotJid], departure_time: datetime):
-        super().__init__(jid, password, PortTerminalOntology.instance())
-        self._slot_manager_agents_jids = slot_manager_agents_jids  # TODO: Replace this line with fetching jids from DF
-        self._departure_time: datetime = departure_time
-        self._slot_id = None
-        self._slot_jid = None
+class SearchSlotManagersHandlerBehaviour(HandleSearchBehaviour):
+
+    async def handleResponse(self, result: Optional[Sequence[DFAgentDescription]]):
+        # pprint(result)
+        self.agent._slot_manager_agents_jids = [SlotJid(int(x.service.properties['slot_id']), str(x.agentName)) \
+                                                for x in result]
 
-    async def setup(self):
         allocation_mt = Template()
```

Comment:
Dlaczego ustawianie podstawowych zachowań agenta jest przeniesione do `handleResponse`?

Reply:
żeby przypadkiem nie wpaść do realokacji itd. przed fetchem managerów

Reply:
Wystarczy dodać w pliku *DFAgent.py* w funkcji `__doFipaRequestClient` na końcu:
```
await handlerBehaviour.join()
```
Wtedy mamy pewność, że wyszukowanie się zakończy zanim dodamy zachowania agenta odpowiedzialne za alokację, realokację i dealokację. Więc setupowanie tych zachowań można przenieść do metody `setup` agenta.

Reply:
bardziej chodziło mi o to żeby nie móc odbierać wiadomości przed fetchem

Reply:
To tak powinno zadziałać rozwiązanie, które zaproponowałam

Reply:
właśnie sprawdziłem i nie działa, może jeśli nie jest to jakiś bardzo duży problem możemy to przepchnąć tak?